drivers/kw2xrf/kw2xrf.c

```diff
@@ -0,0 +1,1041 @@
+/*
+ * Copyright (C) 2015 PHYTEC Messtechnik GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     kw2xrf
+ * @{
+ * @file        kw2xrf.c
+ * @brief       Basic functionality of kw2xrf driver
+ *
+ * @author      Johann Fischer <j.fischer@phytec.de>
+ * @author      Jonas Remmert <j.remmert@phytec.de>
+ * @}
+ */
+#include "crash.h"
+#include "kw2xrf.h"
+#include "kw2xrf_spi.h"
+#include "kw2xrf_reg.h"
+#include "mutex.h"
+#include "msg.h"
+#include "periph/gpio.h"
+#include "net/ng_netbase.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/**
+ * @brief   Internal driver event type in case of an unexpected interrupt
+ */
+#define ISR_EVENT_UNKNOWN      (0x0020)
+
+/* Modem_PA_PWR Register (PA Power Control) has a valid range from 3-31 */
+#define MKW2XDRF_PA_RANGE_MAX      31       /**< Maximum value of PA Power Control Register */
+#define MKW2XDRF_PA_RANGE_MIN      3        /**< Minimum value of PA Power Control Register */
+
+/* PLL integer lookup table */
+static const uint8_t pll_int_lt[16] = {
+    11, 11, 11, 11,
+    11, 11, 12, 12,
+    12, 12, 12, 12,
+    13, 13, 13, 13
+};
+
+/* PLL frequency fractional lookup table */
+static const uint16_t pll_frac_lt[16] = {
+    10240, 20480, 30720, 40960,
+    51200, 61440, 6144, 16384,
+    26624, 36864, 47104, 57344,
+    2048, 12288, 22528, 32768
+};
+
+static const uint8_t pow_lt[44] = {
+    3, 4, 4, 5,
+    6, 6, 7, 7,
+    8, 9, 9, 10,
+    11, 11, 12, 13,
+    13, 14, 14, 15,
+    16, 16, 17, 18,
+    18, 19, 20, 20,
+    21, 21, 22, 23,
+    23, 24, 25, 25,
+    26, 27, 27, 28,
+    28, 29, 30, 31
+};
+
+static const int level_lt[29] = {
+    -35, -34, -32, -31,
+    -29, -28, -26, -25,
+    -23, -22, -20, -19,
+    -17, -16, -14, -13,
+    -11, -10, -8, -7,
+    -5, -4, -2, -1,
+    1, 2, 4, 5,
+    7
+};
+
+void kw2xrf_set_option(kw2xrf_t *dev, uint16_t option, bool state);
+
+int _set_tx_power(kw2xrf_t *dev, int8_t *val, size_t len)
+{
+
+    if (len < 1) {
```

Comment:
```_set_tx_power()``` is called from ```_set(netdev-option)``` which receives an 16bit variable, accordingly the value_len of *val is 2. But to be consistent with the atmel driver I will move this check directly to the netconf_option setter and check for ```<sizeof(uint16_t)```